backends/mediatek/runtime/NeuronBackend.cpp

```diff
@@ -0,0 +1,153 @@
+
+    auto delegate = std::unique_ptr<NeuronExecuTorchDelegate>(
+        new (std::nothrow) NeuronExecuTorchDelegate());
+    if (delegate == nullptr) {
+        return nullptr;
+    }
+    auto res = delegate->LoadCompiledNetwork(Payload, setting);
+    return res == NEURON_NO_ERROR ? delegate.release() : nullptr;
+}
+
+Error NeuronBackend::execute(__ET_UNUSED BackendExecutionContext& context,
+                             DelegateHandle* handle,
+                             EValue** args) const {
+    NeuronExecuTorchDelegate* delegate = reinterpret_cast<NeuronExecuTorchDelegate*>(handle);
+    return delegate->execute(context, args);
+}
+
+void NeuronBackend::destroy(DelegateHandle* handle) const {
+    NeuronExecuTorchDelegate* delegate = reinterpret_cast<NeuronExecuTorchDelegate*>(handle);
+    delete delegate;
+}
+
+bool NeuronBackend::is_available() const {
+    return true;
+}
+
+Error NeuronExecuTorchDelegate::execute(
+      __ET_UNUSED BackendExecutionContext& context,
+      EValue** args) const {
+    if (HintNeuronBackend(args) != NEURON_NO_ERROR) {
+        return Error::InvalidState;
+    };
+
+    auto& allocator = GET_NEURON_ALLOCATOR;
+    size_t inputCount = mInputSizes.size(), outputCount = mOutputSizes.size();
+
+    for (int i = 0; i < inputCount; i++) {
+        auto data_ptr = args[i]->toTensor().data_ptr();
+        auto data_size = args[i]->toTensor().nbytes();
+        if (IsCached</*isInput=*/true>(i, data_ptr)) {
+            continue;
+        };
+        auto unit = allocator.Find(data_ptr);
+        if (unit) {
+            UpdateCache<true>(i, data_ptr);
+            size_t offset = (char*)data_ptr - (char*)unit->GetAddress();
+            mExecutor.SetInputOutputFromMemory</*isInput*/true>(i, unit->GetNeuronMemory(), offset, data_size);
+        } else {
+            mExecutor.SetInputOutput</*isInput=*/true>(i, data_ptr, data_size);
+        }
+    }
+
+    for (int o = inputCount; o < inputCount + outputCount; o++) {
+        auto data_ptr = args[o]->toTensor().data_ptr();
+        auto data_size = args[o]->toTensor().nbytes();
+        auto output_index = o - inputCount;
+        if (IsCached</*isInput=*/false>(output_index, data_ptr)) {
+            continue;
+        };
+        auto unit = allocator.Find(data_ptr);
+        if (unit) {
+            UpdateCache</*isInput=*/false>(output_index, data_ptr);
+            size_t offset = (char*)data_ptr - (char*)unit->GetAddress();
+            mExecutor.SetInputOutputFromMemory</*isInput*/false>(output_index, unit->GetNeuronMemory(), offset, data_size);
+        } else {
+            mExecutor.SetInputOutput</*isInput=*/false>(output_index, data_ptr, data_size);
+        }
+    }
+
+    return mExecutor.Compute() == NEURON_NO_ERROR ? Error::Ok : Error::InvalidState;
+};
+
+int NeuronExecuTorchDelegate::HintNeuronBackend(EValue** args) const {
```

Comment:
'import forever' is an optimization hint designed for specific models including Llama. We prefer not to automatically apply it to every model because it could leads unknow effects.

Reply:
I see - hmm probably add more comments about it (or maybe just copy paste your response)